src/CMakeLists.txt

```diff
@@ -29,7 +29,22 @@ SET (libsolv_HEADERS
     strpool.h dirpool.h knownid.h transaction.h rules.h problems.h
     chksum.h dataiterator.h ${CMAKE_BINARY_DIR}/src/solvversion.h)
 
-SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+IF (NOT MSVC)
+    SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+ELSE ()
+    INCLUDE_DIRECTORIES (./win32_compat)
+    LIST(APPEND libsolv_SRCS
+        win32_compat/fnmatch.c
+        win32_compat/getopt.c
+        win32_compat/pread.c
+        win32_compat/regcomp.c
+        win32_compat/regexec.c
+        win32_compat/strcasecmp.c
+        win32_compat/strcasestr.c
+        win32_compat/tre-mem.c
+    )
+ENDIF ()
```

Comment:
This extra stuff isn't for just MSVC, this is for Win32 as a whole. Do the right thing here, please.

Reply:
Sure, I can change that.